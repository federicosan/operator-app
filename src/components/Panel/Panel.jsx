import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';

// import ConversationPane from '../ConversationPane/ConversationPane.jsx';
// import MessagePane from '../MessagePane/MessagePane.jsx';

export class PanelComponent extends Component {
  static propTypes = {
    children: PropTypes.node,
    dispatch: PropTypes.func,
  }

  constructor (props) {
    super(props);

    this.state = {};
  }

  // getChildContext () {
  //   return { store: this.context.store };
  // }

  render () {
    const { children } = this.props;

    return (
      <div className="panel">
        {children}
      </div>
    );
  }
}

const mapStateToProps = state => ({

});

const mapDispatchToProps = dispatch => ({
  dispatch,
});

const Panel = connect(
  mapStateToProps,
  mapDispatchToProps,
)(PanelComponent);

export default Panel;
