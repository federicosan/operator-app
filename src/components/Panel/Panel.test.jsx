import React from 'react';

import { shallow } from 'enzyme';

import Panel from './Panel.jsx';

const store = {
  subscribe: jest.fn(),
  dispatch: jest.fn(),
  getState: jest.fn(() => ({ })),
};

describe('Panel', () => {
  it('matches snapshot', () => {
    const component = shallow(<Panel store={store} />);

    expect(component).toMatchSnapshot();
  });
});
