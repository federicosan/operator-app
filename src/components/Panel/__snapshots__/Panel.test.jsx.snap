exports[`Panel matches snapshot 1`] = `
<PanelComponent
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "subscribe": [Function],
    }
  } />
`;
